providers/cliflag/cliflag.go

```diff
@@ -0,0 +1,118 @@
+// Package cliflag implements a koanf.Provider that reads commandline
+// parameters as conf maps using ufafe/cli flag.
+package cliflag
+
+import (
+	"strings"
+
+	"github.com/knadh/koanf/maps"
+	"github.com/urfave/cli/v2"
+)
+
+// CliFlag implements a cli.Flag command line provider.
+type CliFlag struct {
+	ctx   *cli.Context
```

Comment:
since [v1](https://github.com/urfave/cli/tree/v1.22.16), [v2](https://github.com/urfave/cli/tree/v2.27.5) are not deprecated, I would say, having support for all active the versions would be best.